confignode/src/main/java/org/apache/iotdb/confignode/client/async/datanode/AsyncDataNodeClientPool.java

```diff
@@ -183,6 +224,31 @@ public void sendAsyncRequestToDataNodeWithRetry(
     }
   }
 
+  public void sendAsyncRequestToDataNodeWithRetry(Object req, AbstractRetryHandler handler) {
+    Map<Integer, TDataNodeLocation> dataNodeLocationMap = handler.getDataNodeLocationMap();
+    if (dataNodeLocationMap.isEmpty()) {
+      return;
+    }
+    for (int retry = 0; retry < MAX_RETRY_NUM; retry++) {
+      CountDownLatch countDownLatch = new CountDownLatch(dataNodeLocationMap.size());
+      handler.setCountDownLatch(countDownLatch);
+      for (TDataNodeLocation targetDataNode : dataNodeLocationMap.values()) {
+        handler.setTargetDataNode(targetDataNode);
+        sendAsyncRequestToDataNode(targetDataNode, req, handler, retry);
```

Comment:
There is concurrent issue here. 

The `targetDataNode` may be changed before every request. When one request's `onComplete` is invoked asynchronously, its `targetDataNode` may not be the real one because it may be changed by following request

Reply:
Fixed. PTAL